tests/unit/dom/helper-hooks-test.js

```diff
@@ -0,0 +1,48 @@
+import { module, test } from 'qunit';
+import { registerHook } from '@ember/test-helpers';
+import { registeredHooks, runHooks } from '@ember/test-helpers/-internal/helper-hooks';
+
+module('helper hooks', function () {
+  test('it can register a hook for a helper', function (assert) {
+    let func = () => {};
+    let hook = registerHook('click', 'start', func);
+
+    let registeredHook = registeredHooks.get('click:start');
+
+    assert.ok(registeredHooks.has('click:start'));
+    assert.equal(registeredHook.size, 2);
+    assert.ok(registeredHook.has(func));
+
+    hook.unregister();
+  });
+
+  test('it can register an unregister a hook for a helper', function (assert) {
+    let func = () => {};
+    let hook = registerHook('click', 'start', func);
+
+    let registeredHook = registeredHooks.get('click:start');
+
+    assert.ok(registeredHooks.has('click:start'));
+    assert.equal(registeredHook.size, 2);
+    assert.ok(registeredHook.has(func));
+
+    hook.unregister();
+
+    assert.notOk(registeredHook.has(func));
+  });
```

Comment:
Instead of testing the implementation details, lets test what we expect folks to do (we don't _really_ care **how** the `runHooks`/`registerHook` implementation works, we care that it works):

```suggestion
  test('it can register an unregister a hook for a helper', function (assert) {
    let func = () => assert.step('click:start hook');
    let hook = registerHook('click', 'start', func);

    // it runs the hook
    runHooks('click', 'start');    
    assert.verifySteps(['click:start hook']);

    // can run multiple times
    runHooks('click', 'start');    
    assert.verifySteps(['click:start hook']);

    // unregister works
    hook.unregister();
    runHooks('click', 'start');    
    assert.verifySteps([]);
  });
```

Can you update all the other tests in this file accordingly?